docs/website/docs/walkthroughs/zendesk-weaviate.md

```diff
@@ -0,0 +1,179 @@
+
+```toml
+[sources.zendesk.zendesk_support.credentials]
+password = "..."
+subdomain = "..."
+email = "..."
+```
+
+### Destination
+
+For the destination we're using [Weaviate Cloud Services](https://console.weaviate.cloud/). You would need to create an account and get a URL and an API key for your Weaviate instance. We're also be using [OpenAI API](https://platform.openai.com/) to generate embeddings for the text data needed to perform vector search. If you haven't already, you would need to create an account and get an API key for OpenAI API.
+
+When you have the credentials, add more lines to the dlt secrets file `~/.dlt/secrets.toml`:
+
+```toml
+[destination.weaviate.credentials]
+url = "https://weaviate_url"
+api_key = "api_key"
+
+[destination.weaviate.credentials.additional_headers]
+X-OpenAI-Api-Key = "sk-..
+```
+
+### Configuring the pipeline
+
+When you run `dlt init zendesk weaviate`, dlt creates a file called `zendesk_pipeline.py` in the current directory. This file contains an example pipeline that you can use to load data from Zendesk source. Let's edit this file to make it work for our use case:
+
+```python
+import dlt
+from dlt.destinations.weaviate import weaviate_adapter
+
+from zendesk import zendesk_support
+
+def main():
+    # 1. Create a pipeline
+    pipeline = dlt.pipeline(
+        pipeline_name="weaviate_zendesk_pipeline",
+        destination="weaviate",
+        dataset_name="zendesk_data",
+    )
+
+    # 2. Initialize Zendesk source to get the ticket data
+    zendesk_source = zendesk_support(load_all=False)
+    tickets = zendesk_source.tickets
+
+    info = pipeline.run(
+        # 3. Here we use a special function to tell Weaviate to
+        # tell Weaviate which fields to vectorize
+        weaviate_adapter(
+            tickets,
+            vectorize=["subject", "description"],
+        ),
+    )
+
+    return info
+
+if __name__ == "__main__":
+    load_info = main()
+    print(load_info)
+```
+
+### Running the pipeline
+
+Now that we have the pipeline configured, we can run it:
+
+```bash
+python zendesk_pipeline.py
+```
+
+We have successfully loaded the data from Zendesk to Weaviate. Let's check it out.
+
+## Query the data
+
+Let's query the Weaviate to select all tickets similar to the ticket with the ID `56b9449e-65db-5df4-887b-0a4773f52aa7` with the subject of the ticket is “How do I change the password for my account?”
+
+```python
+    import weaviate
+    client = weaviate.Client(
+        url='YOUR_WEAVIATE_URL',
+        auth_client_secret=weaviate.AuthApiKey(
+            api_key='YOUR_WEAVIATE_API_KEY'
+        ),
+        additional_headers={
+            "X-OpenAI-Api-Key": 'YOUR_OPENAI_API_KEY'
+        }
+    )
+
+    # Select all tickets similar to the ticket with the ID.
+    response = (
+        client.query
```

Comment:
couldn't we use "vector search" query, not "get"? ie. "problems with passwords" would get a nice list of issues with that topic. if you can order it by metadata ie. updated_at, it would be nice

Reply:
It's actually a vector search as well, it's searching for a "nearest object".

But I can add another example with "with_near_text", e.g.:
```python
response = (
    client.query
    .get("Question", ["question", "answer", "category"])
    .with_near_text({"concepts": ["biology"]})
    .with_limit(2)
    .do()
)